MSBLOC.Web/Program.cs

```diff
@@ -19,7 +22,41 @@ public static void Main(string[] args)
 
         public static IWebHost BuildWebHost(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration(new Program().MSBLOCConfigureAppConfiguration)
                 .UseStartup<Startup>()
                 .Build();
+
+        public void MSBLOCConfigureAppConfiguration(WebHostBuilderContext context, IConfigurationBuilder config)
+        {
+            config.AddEnvironmentVariables("MSBLOC_");
+
+            var builtConfig = config.Build();
+
+            var azureKeyVault = builtConfig["Azure:KeyVault"];
+
+            if (!string.IsNullOrEmpty(azureKeyVault))
+            {
+                var keyVaultConfigBuilder = GetConfigurationBuilder();
+
+                var keyVaultClient = GetKeyVaultClient();
+
+                keyVaultConfigBuilder.AddAzureKeyVault($"https://{azureKeyVault}.vault.azure.net/", keyVaultClient, new DefaultKeyVaultSecretManager());
+
+                var keyVaultConfig = keyVaultConfigBuilder.Build();
+
+                config.AddConfiguration(keyVaultConfig);
+            }
+        }
+
+        protected virtual KeyVaultClient GetKeyVaultClient()
```

Comment:
Are these two functions virtual for any particular reason? Like why aren't they static?

Reply:
For testability. See the "ProgramStub" class in the tests.